packages/flutter/lib/src/widgets/radio_base.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+import 'ticker_provider.dart';
+import 'toggleable.dart';
+import 'widget_state.dart';
+
+/// Signature for [RadioBase.painterGetter]
+///
+/// The getter can use `state` to configure the painter before returning.
+typedef RadioPainterGetter = CustomPainter Function(ToggleableStateMixin state);
```

Comment:
Why make it take a `ToggleableStateMixin` as an argument instead of just a set of WidgetState?

Reply:
It needs additional state like position or downposition and others. we could make into a sets of parameters. maybe that is fine. let me think a bit.